core/src/main/java/org/infinispan/factories/EmptyConstructorNamedCacheFactory.java

```diff
@@ -116,7 +117,12 @@
          } else if (componentType.equals(EvictionManager.class)) {
             return (T) new EvictionManagerImpl();
          } else if (componentType.equals(ExpirationManager.class)) {
-            return (T) new ExpirationManagerImpl();
+            CacheMode cacheMode = configuration.clustering().cacheMode();
+            if (cacheMode.isDistributed() || cacheMode.isReplicated()) {
```

Comment:
I prefer using a dedicated factory when the actual component type depends on the configuration.